packages/contracts-periphery/test/contracts/universal/DrippieRedd.spec.ts

```diff
@@ -0,0 +1,260 @@
+      expect(await DrippieRedd.owner()).to.equal(signer1.address)
+    })
+  })
+
+  describe('create', () => {
+    describe('when called by the owner', () => {
+      it('should create a drip with the given name', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+
+        const drip = await DrippieRedd.drips(DEFAULT_DRIP_NAME)
+        expect(drip.status).to.equal(1)
+        expect(drip.last).to.deep.equal(hre.ethers.BigNumber.from(0))
+        expect(_.toPlainObject(drip.config)).to.deep.include(
+          DEFAULT_DRIP_CONFIG
+        )
+      })
+
+      it('should not be able to create the same drip twice', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+
+        await expect(
+          DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        ).to.be.revertedWith('DrippieRedd: drip with that name already exists')
+      })
+    })
+
+    describe('when called by not the owner', () => {
+      it('should revert', async () => {
+        await expect(
+          DrippieRedd.connect(signer2).create(
+            DEFAULT_DRIP_NAME,
+            DEFAULT_DRIP_CONFIG
+          )
+        ).to.be.revertedWith('UNAUTHORIZED')
+      })
+    })
+  })
+
+  describe('update', () => {
+    describe('when called by the owner', () => {
+      it('should update the config for the given drip', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        await DrippieRedd.update(DEFAULT_DRIP_NAME, {
+          ...DEFAULT_DRIP_CONFIG,
+          recipient: '0x' + '22'.repeat(20),
+        })
+
+        const drip = await DrippieRedd.drips(DEFAULT_DRIP_NAME)
+        expect(drip.status).to.equal(1)
+        expect(drip.last).to.deep.equal(hre.ethers.BigNumber.from(0))
+        expect(_.toPlainObject(drip.config)).to.deep.include({
+          ...DEFAULT_DRIP_CONFIG,
+          recipient: '0x' + '22'.repeat(20),
+        })
+      })
+
+      it('should revert if the drip does not exist yet', async () => {
+        await expect(
+          DrippieRedd.update(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        ).to.be.revertedWith('DrippieRedd: drip with that name does not exist')
+      })
+    })
+
+    describe('when called by not the owner', () => {
+      it('should revert', async () => {
+        await expect(
+          DrippieRedd.connect(signer2).update(
+            DEFAULT_DRIP_NAME,
+            DEFAULT_DRIP_CONFIG
+          )
+        ).to.be.revertedWith('UNAUTHORIZED')
+      })
+    })
+  })
+
+  describe('toggle', () => {
+    describe('when called by the owner', () => {
+      it('should toggle the status of a drip', async () => {
+        // On by default
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        expect((await DrippieRedd.drips(DEFAULT_DRIP_NAME)).status).to.equal(1)
+
+        // Toggle off
+        await DrippieRedd.toggle(DEFAULT_DRIP_NAME)
```

Comment:
also wait on the toggle tx to avoid flakes.

Reply:
Same comment as above here, doing `await` on the tx is good enough since it gets executed within the `await`